src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -127,25 +127,34 @@ public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToJSON(DataType targetCQ
   }
 
   // Boolean
-  public static final JSONCodec<Boolean, Boolean> BOOLEAN =
+  private static final JSONCodec<Boolean, Boolean> BOOLEAN =
       new JSONCodec<>(
           GenericType.BOOLEAN,
           DataTypes.BOOLEAN,
           JSONCodec.ToCQL.unsafeIdentity(),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::booleanNode));
 
   // Numeric Codecs
-  public static final JSONCodec<BigDecimal, Long> BIGINT =
+  private static final JSONCodec<BigDecimal, Long> BIGINT_FROM_BIG_DECIMAL =
       new JSONCodec<>(
           GenericType.BIG_DECIMAL,
           DataTypes.BIGINT,
           JSONCodec.ToCQL.safeNumber(BigDecimal::longValueExact),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
 
-  // TODO Tatu For performance reasons we could also consider only converting FP values into
-  // BigDecimal JsonNode -- but converting CQL integer values into long-valued JsonNode.
-  //  I think our internal handling can deal with Integer and Long valued JsonNodes and this avoids
-  // some of BigDecimal overhead (avoids conversion overhead, serialization is faster).
+  private static final JSONCodec<BigInteger, Long> BIGINT_FROM_BIG_INTEGER =
```

Comment:
I get what we are doing with integral values from Jackson - question is what need to support both BigInteger and Long source values ? if it is needed can we add a comment at top to explain strategy  . 

And that the type mappings here must sync with how we pull values from Jackson.

Reply:
So, with FPs, cost of processing `BigDecimal` seems to be roughly similar to `Double`.
But cost of processing `BigInteger` is significantly higher than that of boxed primitive integral types. So I figured it is a compromise to try to avoid huge number of converters by slight addition -- I think minimally we could go with `BigDecimal` and `BigInteger`.

I'll add a comment here.

100% on mentioning that handling with `RowShredder` must be kept in sync.

Reply:
I added some class Javadoc comments in `RowShredder` and `JSONCodec`, hope that suffices.